Microsoft.Toolkit.Uwp.UI.Controls/TabView/TabView.cs

```diff
@@ -0,0 +1,395 @@
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Linq;
+using Microsoft.Toolkit.Uwp.UI.Extensions;
+using Windows.ApplicationModel.DataTransfer;
+using Windows.Foundation;
+using Windows.Foundation.Collections;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Controls.Primitives;
+
+namespace Microsoft.Toolkit.Uwp.UI.Controls
+{
+    /// <summary>
+    /// TabView is a control for displaying a set of tabs and their content.
+    /// </summary>
+    [TemplatePart(Name = TABCONTENTPRESENTER_NAME, Type = typeof(ContentPresenter))]
+    [TemplatePart(Name = TABVIEWCONTAINER_NAME, Type = typeof(Grid))]
+    [TemplatePart(Name = TABITEMSPRESENTER_NAME, Type = typeof(ItemsPresenter))]
+    [TemplatePart(Name = TABSCROLLVIEWER_NAME, Type = typeof(ScrollViewer))]
+    [TemplatePart(Name = TABSCROLLBACKBUTTON_NAME, Type = typeof(ButtonBase))]
+    [TemplatePart(Name = TABSCROLLFORWARDBUTTON_NAME, Type = typeof(ButtonBase))]
+    public partial class TabView : ListViewBase
+    {
+        private const int SCROLL_AMOUNT = 50; // TODO: Should this come from TabWidthProvider?
+
+        private const string TABCONTENTPRESENTER_NAME = "TabContentPresenter";
+        private const string TABVIEWCONTAINER_NAME = "TabViewContainer";
+        private const string TABITEMSPRESENTER_NAME = "TabsItemsPresenter";
+        private const string TABSCROLLVIEWER_NAME = "TabsScrollViewer";
+        private const string TABADDBUTTON_NAME = "AddTabButton";
+        private const string TABSCROLLBACKBUTTON_NAME = "ScrollBackButton";
+        private const string TABSCROLLFORWARDBUTTON_NAME = "ScrollForwardButton";
+
+        private ContentPresenter _tabContentPresenter;
+        private Grid _tabViewContainer;
+        private ItemsPresenter _tabItemsPresenter;
+        private ScrollViewer _tabScroller;
+        private ButtonBase _tabAddButton;
+        private ButtonBase _tabScrollBackButton;
+        private ButtonBase _tabScrollForwardButton;
+
+        /// <summary>
+        /// Occurs when a tab is dragged by the user outside of the <see cref="TabView"/>.  Generally, this paradigm is used to create a new-window with the torn-off tab.
+        /// The creation and handling of the new-window is left to the app's developer.
+        /// </summary>
+        public event EventHandler<TabDraggedOutsideEventArgs> TabDraggedOutside;
+
+        /// <summary>
+        /// Occurs when a tab's Close button is clicked.  Set <see cref="TabClosingEventArgs.Cancel"/> to true to prevent automatic Tab Closure.
+        /// </summary>
+        public event EventHandler<TabClosingEventArgs> TabClosing;
+
+        /// <summary>
+        /// Occurs when a the Add button is clicked (visible when <see cref="IsAddTabButtonVisible"/> is true).
+        /// </summary>
+        public event EventHandler AddTabButtonClick;
```

Comment:
Remove along with the IsAddTabButtonVisible property in favor of a more flexible model that allows the user to put their own content in it's place. NOTE: That support is already there, but just make it the only way to have an add button.

Reply:
The add scenario is going to be the most common though, so having it be a default hook seems important.  The user can indeed overwrite it if they want.  I should expose a command property though.

Reply:
Adding Tabs is the minority use case. Majority will be to have a static amount of tabs. Think of all the applications you've used that have tabs. How many have a way to add a new tab? Browsers and....

Reply:
Moved to the sample outside the template.